build-release.sh

```diff
@@ -47,7 +49,7 @@ sign_macos() {
             codesign --force --timestamp \
               --options=runtime --entitlements editor.entitlements \
               -s ${OSX_KEY_ID} -v ${_appname} && \
-            zip -r ${_binname}_signed.zip ${_appname}"
+            $ZIP -r ${_binname}_signed.zip ${_appname}"
```

Comment:
So this bit forces me to have 7zip installed on macOS, which is less trivial than on Linux.

It seems available on homebrew but homebrew isn't set up on HP's Mac, could likely install it.

But this begs the question of whether there's anything more actively maintained than 7zip to compress zips. The last release dates back to 2016.

Reply:
> But this begs the question of whether there's anything more actively maintained than 7zip to compress zips. The last release dates back to 2016.

The last GUI release was made in June 2023: https://7-zip.org/
However, the CLI version was never updated past 16.02 (May 2016). This is an unfortunate artifact of 7-zip being developed around its Windows GUI first, and also being developed behind closed doors.

I don't know of any alternatives that have the same level of efficiency and performance for compressing ZIP archives. Using Zopfli to create a ZIP archive will result in slightly more efficiency, but it's more than 10× slower which makes it not worth the time.

Reply:
> The last GUI release was made in June 2023: [7-zip.org](https://7-zip.org/)

It's Windows only though, no?

I found PeaZip which seems actively maintained and properly cross-platform: https://github.com/peazip/PeaZip/releases/

Might be worth comparing with 7zip.

Reply:
> But this begs the question of whether there's anything more actively maintained than 7zip to compress zips. The last release dates back to 2016.

At least of macOS homebrew, there are two 7Z packages:
- `p7zip` - `7z` executable, built from p7zip port source from 2017 (17.04).
- `sevenzip` - `7zz` executable, built from official 2023 source (23.01, same as GUI version).

Seems like official source added support for Linux/macOS command line build in 2021.